validator/client/validator.go

```diff
@@ -1139,7 +1140,40 @@ func (v *validator) StartEventStream(ctx context.Context, topics []string, event
 	v.validatorClient.StartEventStream(ctx, topics, eventsChannel)
 }
 
-func (v *validator) ProcessEvent(event *eventClient.Event) {
+func (v *validator) checkDependentRoots(ctx context.Context, head *structs.HeadEvent, slot primitives.Slot) error {
+	if head == nil {
+		return errors.New("received empty head event")
+	}
+	prevDepedentRoot, err := bytesutil.DecodeHexWithLength(head.PreviousDutyDependentRoot, fieldparams.RootLength)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode previous duty dependent root")
+	}
+	currEpochStart, err := slots.EpochStart(slots.ToEpoch(slot))
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(prevDepedentRoot, v.duties.PrevDependentRoot) {
+		if err := v.UpdateDuties(ctx, currEpochStart); err != nil {
+			return errors.Wrap(err, "failed to update duties")
+		}
+		log.Info("Updated duties due to previous dependent root change")
```

Comment:
If we're going to log here, I'd prefer we add more detail to the log. It could include `prevDependentRoot`, `v.duties.PrevDependentRoot`, the slot, or anything else that adds verbosity to help with debugging.

Reply:
+1, lets add a bit more detail

Reply:
this is already logged in the beacon, I don't see any reason to add this log.

Reply:
dont feel strongly here but keep in mind beacon and VC can be run under different entities